### ISSUE TITLE
p-overlayPanel in a position:sticky div in a scrollable container has bad position

### ISSUE THREAD

**Issue by Ldoppea** [primefaces/primeng#5086]
Hi,
I just tried to add a p-overlayPanel in a div with a sticky behavior. This div is inside a scrollable container so `appendTo="body"` cannot be use (or the overlay won't follow the scrolled content).

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
```

**Plunkr Case (Bug Reports)**
http://plnkr.co/edit/20NH9GgGoeLQQzsTYmhn?p=preview

**Current behavior**
The overlayPanel is wrongly positioned.
If I add `appendTo="stickyRef"` nothing changes.
If I add `appendTo="body"` it is well positionned BUT it does not follow the scroll.

**Expected behavior**
It should be positioned right under the button and follow the scroll.

**Minimal reproduction of the problem with instructions**
http://plnkr.co/edit/20NH9GgGoeLQQzsTYmhn?p=preview
Click on `Basic overlay`, the overlayPanel appears far from the button.
add `appendTo="body"` to the overlayPanel.
Click on `Basic overlay`, the overlayPanel appears right under the button.
Scroll the div, the overlayPanel doesn't stick to the button.

**Please tell us about your environment:**
Windows 7, Google Chrome

* **Angular version:** 5.1.1

* **PrimeNG version:** 5.0.2

* **Browser:** [Chrome 63]
 
* **Language:** [TypeScript 2.6.1]

**Comment by cagataycivici** [comment]
You need to create a wrapper div that contains the button and the overlay panel and append the overlap to this div. Tooltip documentation has an example.

**Comment by Kiiiwiii** [comment]
@cagataycivici 
It does not solve the issue: 
I follow the examples from tooltips, weird behavior is still there.
```
<div #overlayAnchor  style="display:inline-block;position:relative">
    <button pButton
            (click)="overlayPanel.show($event)"
            [label]="abc"
            ></button>
    <p-overlayPanel #overlayPanel [appendTo]="overlayAnchor">
         someContent
    </p-overlayPanel>
</div>
```

The position of the overlay panel is incorrectly calculated !

**Comment by jayscheidt** [comment]
I am seeing the same behavior as @Kiiiwiii

**Comment by khushboojh** [comment]
was there a solution?? I am having the same issue. I tried all the solutions mentioned above.

**Comment by salarenko** [comment]
Unfortunately I also struggle with the same issue. I've tried solution recommended by @cagataycivici but with no result. Another option is appending to body, but I'm using overlay-panel inside scrollable dialog, so while scrolling overlay stays in the same position.

**Comment by bartlomiejborzucki** [comment]
Please reopen it. I have the same problem.

**Comment by behroozbc** [comment]
i have same problem with overlayPanel 
Please reopen it 
@cagataycivici 
![image](https://user-images.githubusercontent.com/20381616/71416041-0a5ab500-2674-11ea-8375-d6f02355cacb.png)

**Comment by BhupathiReddyGaddam** [comment]
I have same problem and tried a day to fix it. 
I tried above solutions as well, nothing worked.
Please reopen it and please provide a solution.

**Comment by rhafiko** [comment]
same issue here. 
Please reopen it, let's find a solution.

**Comment by BhupathiReddyGaddam** [comment]
Thank you! @rhafiko

**Comment by rajjaiswalsaumya** [comment]
Any fix for this ?

**Comment by lapalus** [comment]
Yes are there any fix for this?

**Comment by rajjaiswalsaumya** [comment]
Why is this closed ? Is this fixed?

**Comment by dev4ndy** [comment]
Has anyone been able to solve it ? I have the same problem

**Comment by yahia-salah** [comment]
I solved this by setting the [appendTo]="null"

**Comment by abhijit-chikane** [comment]
It is not working for me
with sticky div the panel should remain on that position,no?

**Comment by gtrrz-victor** [comment]
I had the same issue and found out that the problem was how and where I was setting the max height and the overflow. 

In my case, I set the html, body, and React root element to height: 100vh and applied the overflow to the React root element. Then, the overlay panel gets attached at the body level, in other words, outside of the React root element. So, when I scrolled within the root element, the overlay panel was not following the scroll, which is normal since it is outside the scrollable content. I created two minimal examples using the overlay panel: one with a working scroll and another with a broken scroll. 

I hope this helps you somehow.

working example: https://stackblitz.com/edit/aagajg
broken example: https://stackblitz.com/edit/9blmbd-dl75vh
